src/js/port_handler.js

```diff
@@ -29,6 +32,94 @@ PortHandler.initialize = function () {
     // fill dropdown with version numbers
     generateVirtualApiVersions();
 
+    const self = this;
+
+    self.mdnsBrowser = {
+        services: [],
+        browser: null,
+        init: false,
+    };
+
+    let bonjour = {};
+
+    if (!self.mdnsBrowser.init) {
+        if (GUI.isCordova()) {
+            const zeroconf = cordova.plugins.zeroconf;
+            zeroconf.registerAddressFamily = 'ipv4'; // or 'ipv6' ('any' by default)
+            zeroconf.watchAddressFamily = 'ipv4'; // or 'ipv6' ('any' by default)
+            zeroconf.watch("_http._tcp.", "local.", (result) => {
+                const action = result.action;
+                const service = result.service;
+                if (['added', 'resolved'].includes(action)) {
+                    console.log("Zeroconf Service Changed", service);
```

Comment:
Would any of these logs be useful for the UI layer? E.G. https://github.com/betaflight/betaflight-configurator/blob/master/src/js/msp/MSPHelper.js#L2940

Reply:
totaly, i will change it, so that those logs go to the GUI as well.

Reply:
Please change in follow up patch.